### PR TITLE
Mothership vault gear entries in supply console

### PR DESCRIPTION
**Pull request by ManopolisJones**
## Changes
Gives various items from the mothership vault their own entries in the shuttle supply console

- Adds mothership uniform bin under contraband clothing group for 50 credits (25 for each item in crate), will give a random set of either two laborer coveralls, two explorer uniforms, or a science uniform and mothership labcoat
- Adds MDF Surplus weapons crate under hidden security group for 60 credits (double the security weapons crate), will give two stunprobes and two disintegrators
- Adds MDF Surplus standard armor bin under contraband security group for 40 credits (double the armor crate), gives two standard mothership body armors, two standard mothership helmets, and two grey soldier jumpsuits

MDF Surplus weapon and armor crate are both security access locked currently.

## Elaboration
Always thought it was unfortunate that all the cool new mothership gear was locked behind an increasingly rare vault. So I figured it was a good idea to make their own entries to the e-magged and hacked cargo supply console, that way they have a chance to see more action.

I attempted to model the supply entries off of equivalent normal cargo entries, for example the MDF standard armor bin was modeled after the normal armor crate, or the Mothership uniform crate after the Grey Space-Ex crate.

In choosing a bin for the armor and uniform crate, I figured it would make for an interesting change, since no other entry uses bins (for the reason likely being their inability to be used with wrapping paper), and I figured lacking my spriting ability, this would make a fairly alien change to the entry (ayys using bins to transport clothing). 

In closing there's a few things still on my mind in regards to this. One, I wish to add custom crates sprites for all the mothership entries, which might be addressed in a future PR regarding crate types. Two, I thought about adding the space polyp as a normal purchasable cattle entry under the Hydroponics group, but I didn't want to add a new entry to largecrate.dm so that may also be addressed in a different PR. 

Finally I wanna know if people would prefer this PR the way it is, or to move some contraband entries to hidden entries and vise versa. Like, should both surplus grenades and weapons be under hidden and both surplus standard and heavy armor be under contraband? Should all MDF Surplus items be under hidden? Tell me what you think, and also thanks to Optimism333 for his great mothership lab vault.

## Images
Weapons crate, standard large reinforced box (color referance), clothing bin, armor bin
noxes

boxes
Not totally married to the clothing bin, will gladly switch it to the normal grey bin, or without the text if people think it would be better
## Changelog

:cl:
 * rscadd: Adds mothership gear entries to e-magged and hacked cargo consoles

### CODE REVIEW

**Comment by pedr69** [comment]
i'm guessing the hidden section is gotten via soldering the console?
and emagging it unlocks both the hidden and contraband stuff?

**Comment by jwhitak** [comment]
Contraband is the soldered board, hidden is emagged. I'm down for more Grey stuff being orderable, but I've got minor concerns about the steel toed boots.

**Comment by Kurfursten** [comment]
👎 for infinite RD boots. This is the problem with making a new toy for heads of staff. First you make something to incentivize playing heads, then someone creeps it to a rare vault item, then someone says "hey instead of making them rare, what if we just let you order as many as you want?"

I'm pretty divided on the subject of creeping so much special vault loot to the main station right away too, given that it was Optimism who went to the trouble of making all that special loot. I'd really like to hear what he has to say, given that this makes a visit to Lab Z12 much less special. Suddenly what was rare, exotic loot is just everyday cargo stuff.

**Comment by djkramer123** [comment]
Orderable steel toed shoes is pretty lame, I don't care much about the other stuff(orderable poly acid grenades is a bit of an eye roll though) but I can see it devaluing actually going to the vault

**Comment by WindowsErrors** [comment]
> -1 for infinite RD boots. This is the problem with making a new toy for heads of staff. First you make something to incentivize playing heads, then someone creeps it to a rare vault item, then someone says "hey instead of making them rare, what if we just let you order as many as you want?"
> 
> I'm pretty divided on the subject of creeping so much special vault loot to the main station right away too, given that it was Optimism who went to the trouble of making all that special loot. I'd really like to hear what he has to say, given that this makes a visit to Lab Z12 much less special. Suddenly what was rare, exotic loot is just everyday cargo stuff.

soon enough next thing you know there's a m*chanic leaving 20 of them in the hallway

**Comment by wotisjanitor** [comment]
Just remove the steel toes

**Comment by Optimism333** [comment]
## Muh Wall of Feedback ##

Oh man, I have thoughts on this one. First off, I appreciate that you took the time to mess with this and that you think my code is worth making more accessible. However, after looking at all this and the comments, I'm inclined to agree that this pr has some problems.

> 👎 for infinite RD boots. 

This is fair, and frankly this wasn't what I intended when I made the regular mothership boots, to make an ordinary ayy vault clothing item a copy of a rare RD clothing item. To fix this problem, I would suggest making the regular ones a child of normal jackboots. I'll fix it myself if that helps.

> * Adds mothership uniform crate under contraband clothing group for 50 credits (25 for each item in crate), will give a random set of either two laborer coveralls, two explorer uniforms, or a science uniform and mothership labcoat

I think this is the part of the pr I am the most comfortable with. Having ayy contraband uniforms that were orderable was something that I fully intended to write up at some point. I wanted to give ayys on the station the opportunity to have a taste of their homegrown fashion (though it also fits humans as well), and balance-wise the regular uniforms (leader uniform being an exception) provide very minor protection in a relevant stat, or are identical to station jumpsuits. Same goes for the mothership labcoat, it's identical to a normal labcoat except slightly slimmed in the center of the sprite to pair better with the ayy researcher jumpsuit.

If the normal mothership boots are made a child of regular jackboots maybe you can also add a few to this crate.

> * Adds MDF Surplus grenade crate under hidden security group for 300 credits (75 for each grenade), will give four random grenades chosen from either the mothership drone grenade (low chance) or the mothership acid grenade (higher chance)

I'm not sure I like this one, after some thinking. The acid grenade could maybe better fit as a relatively expensive addition to the traitor uplink section, it is pretty destructive. Especially when you consider that now they melt everything off a poor sap caught in the radius, not just his helmet and mask.

I would personally prefer to keep the ufo grenades rare for now (partly because the drone projectile still has a tendency to runtime that was never resolved).

> * Adds MDF Surplus heavy armor crate under hidden security group for 120 credits (double the riot gear crate), gives two mothership heavy armors, two mothership heavy helmets, and two MDF belts

I understand that this would be locked behind an emag, but even so, I don't think I want MDF heavy armor to be available through cargo shipping. It's very beefy in both melee and anti-laser, the only drawbacks being that it's not spess proof and has high slowdown. Plus I would accept the argument here that there should be some incentive to visit the vault if you want the very best loot, and the heavy armor is one of the better loot items in the vault.

> * Adds MDF Surplus weapons crate under contraband security group for 60 credits (double the security weapons crate), will give two stunprobes and two disintegrators

I would lock this behind the hidden security group instead (requiring an emag), and have it be more of a counterpart to the high-tech energy weapons crate. Have it contain two disintegrators, and maybe drop the price to 50 credits or leave it as is. 

I'd rather the stun probes stayed rare for now.

> * Adds MDF Surplus standard armor crate under contraband security group for 40 credits (double the armor crate), gives two standard mothership body armors, two standard mothership helmets, and two grey soldier jumpsuits

I don't think this is troublesome for the most part. The armor is identical to sec armor and helmets in stats, and the uniforms just swap 10 melee protection for 10 laser protection and 5 energy protection (which latter of which is one of the rarest damage types in-game). Having this under normal contraband isn't a huge deal to me, as its on the lower end of the vault's loot.

> In closing there's a few things still on my mind in regards to this. One, I wish to add custom crates sprites for all the mothership entries, which might be addressed in a future PR regarding crate types. Two, I thought about adding the space polyp as a normal purchasable cattle entry under the Hydroponics group, but I didn't want to add a new entry to largecrate.dm so that may also be addressed in a different PR.

Couple thoughts on the above. I would absolutely love if you'd make custom sprites for the crates, or found someone who was willing to. Hell, I'd add them to the vault right away if you did. Every little touch like that helps the atmosphere.

As for the spess polyps, there's some things I want to do before they become more widely available. Mainly I wanted to add a way to breed them, and second I wanted to make some unique dishes that were meat/gelatin based. Possibly also some more unique behaviors if I get that far...

But please do reach out to me if you want to help with any of that stuff, I'm not territorial about my code, and I'm very happy to have help and listen to other ideas.

## Conclusion ##
Tried to be as fair as I could with my feedback, and I'm quite happy that you think my mothership lab code is worth making more available. I hope by applying some of the suggestions I made it lessens the worries that it cheapens the value of the vault and also prevents making some of the more valuable/rare items in-game common.

One note. In the future if you want to discuss changes/ideas you have for prs with me, please consider reaching out to me through a github discussion or another communication medium that isn't in-game. It's hard for me to give you useful feedback when I'm distracted by trying to think of a tator gimmick or other in-game shenanigans.

That's pretty much all I've got to say on this.

**Comment by ManopolisJones** [comment]
Alright so, mothership boots crate, grenade crates and heavy armor crates to be btfo'd, mothership standard surplus to be moved to hidden, few crates that remain should be given custom sprites, and space polyps to be given their own entry in another PR when they get updated.

**Comment by Kurfursten** [comment]
I can only speak for myself but I am glad you're treating Optimism's feedback with respect

If you need help with spritework you may be able to ping @Spriters; if you have trouble with the code end you can ping me, I'm happy to help. I do recommend you join the coding Discord so you don't get left out of less formal discussions.

**Comment by ManopolisJones** [comment]
new crates


New crates, what do you think?

**Edit:** Not sure which of these designs for the gear crate I prefer

hmms

**Comment by WindowsErrors** [comment]
hey I know you already went through all the trouble of making the funny black crate with a skull on it but I had an idea

aren't hexagonal honeycombs the most space efficient shape or something? they're the most symmetrical shape that has the least surface area, so a hexagon shaped crate would be the mathematically least likely to cave or deform or bend or otherwise potentially compromise the contents of the crate or something like that

**Comment by Optimism333** [comment]
> new crates
> 
> New crates, what do you think?

Those aren't unacceptable from my view, and I'd be okay with those.

However, if you're interested in pushing yourself a bit, I asked a spriter pal who helped advise me on improving the hoverdisc drone sprite, and he had this to say:

spriteadvice

I think I'd agree from a more critical perspective that the gear crate is a bit too dark. It could use a little more contrast between the dark interior and the sides. Lighten the black and pair it with a nice highlighting color like bright green or such to make it pop, or maybe consider making the crates look more like grey and blue steel, similar to the material used on the hoverdisc drone.

Also another nerd on the coder discord is yelling that the ayy crates should use purple and yellow for locked and open respectively, to be different from standard crates. That one is a very minor thing though, I don't mind it much myself.

> hey I know you already went through all the trouble of making the funny black crate with a skull on it but I had an idea
> 
> aren't hexagonal honeycombs the most space efficient shape or something? they're the most symmetrical shape that has the least surface area, so a hexagon shaped crate would be the mathematically least likely to cave or deform or bend or otherwise potentially compromise the contents of the crate or something like that

This is also a fun suggestion.

**Comment by Optimism333** [comment]
One other idea, btw. I wonder if you couldn't make it so you can use either security access OR mothership military access to open the crates. That would be a neat touch, on the off chance the player has a mothership soldier ID.

**Comment by ManopolisJones** [comment]
> One other idea, btw. I wonder if you couldn't make it so you can use either security access OR mothership military access to open the crates. That would be a neat touch, on the off chance the player has a mothership soldier ID.

Great idea on that one, I think that's totally possible and i'll see about updating the access on the crates once I figure out the sprites I wanna go with

Speaking of the sprites, I am not an artist at all, I simply do alright to edit shit to look different, that being said, i'll take a crack at making the supply gear crate hexagonal. Alternatively i'll make it so the mothership uniform crate and surplus armor crate use a unlocked and locked re-sprited bin crate respectively if I can manage to make that work and look good in the process.

**Comment by Kurfursten** [comment]
Yes, all you have to do is change `access` to `one_access` and list both accesses. The one you want is `access_mothership_military`.

**Comment by Optimism333** [comment]
> Speaking of the sprites, I am not an artist at all, I simply do alright to edit shit to look different, that being said, i'll take a crack at making the supply gear crate hexagonal. Alternatively i'll make it so the mothership uniform crate and surplus armor crate use a unlocked and locked re-sprited bin crate respectively if I can manage to make that work and look good in the process.

No worries. I'm hardly an artist myself, as I'm sure you've seen from the amateur spritework I've been putting out over the past year. The good ones I'm managed to crank out needed time, references, and a lot of feedback.

I'm sure you'll do the best you can.

**Comment by Kurfursten** [comment]
Yes, yes, HOWEVER: terrifying saucer crates (lock colors on bottom)

image

File: code/datums/supplypacks.dm [hunk 1]
```diff
@@ -1027,9 +1027,9 @@ var/list/all_supply_groups = list("Supplies","Clothing","Security","Hospitality"
 					/obj/item/weapon/gun/energy/smalldisintegrator,
 					/obj/item/weapon/gun/energy/smalldisintegrator)
 	cost = 60
-	containertype = /obj/structure/closet/crate/secure/weapon
+	containertype = /obj/structure/closet/crate/secure/ayyweapon
 	containername = "MDF standard weapons crate"
-	access = list(access_armory)
+	one_access = list(access_mothership_military)
```
Comment (Kurfursten) [inline code/datums/supplypacks.dm]: the way it should look is this:
`one_access = list(access_armory, access_mothership_military)`

This tells the game that *either* of these two accesses will open it

**Comment by WindowsErrors** [comment]
a hexagonal or a cylindrical shipping crate seems sufficiently alien, it's something that human engineers wouldn't normally do, we'd do something cheap and blocky, because it's not the most effective products that see the light of day, it's the most efficient

like this
image

I remembered those hexagonal prism containers in Xenonauts that the alien Landing Ship carries, I thought about how they'd be really easy to stack on top of each other but really annoying to actually make

File: code/datums/supplypacks.dm (hunk 1 above)
Comment (ManopolisJones) [reply code/datums/supplypacks.dm]: ah, okay

**Comment by ManopolisJones** [comment]
> a hexagonal or a cylindrical shipping crate seems sufficiently alien, it's something that human engineers wouldn't normally do, we'd do something cheap and blocky, because it's not the most effective products that see the light of day, it's the most efficient
> 
> like this image
> 
> I remembered those hexagonal prism containers in Xenonauts that the alien Landing Ship carries, I thought about how they'd be really easy to stack on top of each other but really annoying to actually make

I think somebody who can actually sprite should return to this idea, and do the ayy storage crates justice.

**Comment by Optimism333** [comment]
> a hexagonal or a cylindrical shipping crate seems sufficiently alien, it's something that human engineers wouldn't normally do, we'd do something cheap and blocky, because it's not the most effective products that see the light of day, it's the most efficient
> 
> like this image
> 
> I remembered those hexagonal prism containers in Xenonauts that the alien Landing Ship carries, I thought about how they'd be really easy to stack on top of each other but really annoying to actually make

That's one the games I looked at for references when I was looking for funny alien crates. I might try my hand at a few concepts myself. General alien-looking structures/furniture would be neat for decoration.

**Review by Kurfursten — APPROVED** [review]

**Comment by WindowsErrors** [comment]
> That's one the games I looked at for references when I was looking for funny alien crates. I might try my hand at a few concepts myself. General alien-looking structures/furniture would be neat for decoration.

the look on the caesan officers face when two men wearing nothing but cyan blue jumpsuits barge into their bridge and instantly whack him five times in the groin with stun batons

**Comment by Kurfursten** [comment]
I like your passion for cargo Manopolis

Looking forward to seeing if you have any bigger content in mind for the future.

**Pull request merged**